src/error.rs

```diff
@@ -325,18 +325,21 @@ impl<I: fmt::Display + Hash + Eq, S: Span> fmt::Display for Simple<I, S> {
                     None => format!("end of input"),
                 },
             )?,
-            _ => write!(
-                f,
-                " but one of {} was expected",
-                self.expected
-                    .iter()
-                    .map(|expected| match expected {
-                        Some(x) => format!("'{}'", x),
-                        None => format!("end of input"),
-                    })
-                    .collect::<Vec<_>>()
-                    .join(", ")
-            )?,
+            _ => {
+                write!(
+                    f,
+                    r"found {:?}, but one of {:?} was expected",
+                    found,
+                    self.expected
+                        .iter()
+                        .map(|expected| match expected {
+                            Some(x) => format!("{}", x),
```

Comment:
don't you need `{:?}` here ?

Reply:
This is the tricky bit. I played around with it for a bit and you can use `{:?}` here, but you'll get very strange output. The escape `\` tokens are included in the format string.

There seems to be several different ways one could do this. I want to introduce a solution that doesn't involve messing with trait bounds or anything else. I'm sure there is a more elegant and/or more optimized way to handle this.

Reply:
I mean, did you try to put `{:?}` here and remove leave a simple `{}` line 331 for the `.. one of {} was expected`

otherwise I guess it'll use the debug output of the debug output, leading to many `\`

Reply:
I'll take a closer, more careful look and implement things as per your suggestions. I may have missed something crucial because my intuition is telling me that your approach is correct.